packages/contracts-bedrock/test/invariants/symbolic/MockL2ToL2Messenger.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import "src/L2/L2ToL2CrossDomainMessenger.sol";
+import "forge-std/Test.sol";
+
+contract MockL2ToL2Messenger is IL2ToL2CrossDomainMessenger {
+    uint256 internal messageNonce;
+    address internal immutable CURRENT_XDOMAIN_SENDER;
+
+    constructor(address _currentXDomSender) {
+        // Setting the current cross domain sender for the check of sender address equals the supertoken address
+        CURRENT_XDOMAIN_SENDER = _currentXDomSender;
+    }
+
+    function sendMessage(uint256 _destination, address _target, bytes calldata) external payable {
+        // TODO: Disable checks?
+        if (_destination == block.chainid) revert MessageDestinationSameChain();
+        if (_target == Predeploys.CROSS_L2_INBOX) revert MessageTargetCrossL2Inbox();
+        if (_target == Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER) revert MessageTargetL2ToL2CrossDomainMessenger();
+    }
+
+    function relayMessage(
+        uint256,
+        uint256,
+        uint256,
+        address,
+        address _target,
+        bytes calldata _message
+    )
+        external
+        payable
+    {
+        // TODO: Add checks?
+
+        // TODO: Not sure if this is needed at all if halmos will be used for stateful tests. But will remain for now.
```

Comment:
let's try to end with a common implementation (now or later, low prior)

Reply:
For Medusa and halmos you mean?